Eliminate use of getPropertyCSSValue from tests, part 1

darinadler opened the pull request:
#### 83c6cf35ee3de8fea68e6bef6f077a5106a7d3f4
<pre>
Eliminate use of getPropertyCSSValue from tests, part 1
<a href="https://bugs.webkit.org/show_bug.cgi?id=293570">https://bugs.webkit.org/show_bug.cgi?id=293570</a>
<a href="https://rdar.apple.com/152021244">rdar://152021244</a>

Reviewed by Anne van Kesteren.

* LayoutTests/animations/fill-mode-reverse.html: Use parseFloat and getPropertyValue.

* LayoutTests/animations/spring-computed-style-expected.txt: Updated.
* LayoutTests/animations/spring-computed-style.html: Use getPropertyValue and js-test.js.

* LayoutTests/animations/spring-parsing-expected.txt: Updated.
* LayoutTests/animations/spring-parsing.html: Removed unnecessary test of getPropertyCSSValue.
Use shouldBeEqualToString and js-test.js.

* LayoutTests/animations/unprefixed-properties-expected.txt: Updated.
* LayoutTests/animations/unprefixed-properties.html: Removed unnecessary test of
getPropertyCSSValue. Use shouldBeEqualToString and js-test.js.

* LayoutTests/animations/unprefixed-shorthand-expected.txt: Updated.
* LayoutTests/animations/unprefixed-shorthand.html: Use getPropertyValue,
shouldBeEqualToString, and js-test.js.

* LayoutTests/css3/calc/transitions-dependent.html: Renamed function in test to
propertyValue instead of getPropertyCSSValue: it has no relationshiop with the
getPropertyCSSValue we are trying to get rid of.

* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled.html: Removed unnecessary
test of getPropertyCSSValue.

* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled.html: Removed unnecessary
test of getPropertyCSSValue.

* LayoutTests/css3/filters/backdrop/backdropfilter-property-computed-style-expected.txt
Updated.
* LayoutTests/css3/filters/backdrop/backdropfilter-property-computed-style.html:
Use getPropertyValue, shouldBeEqualToString, and js-test.js.

* LayoutTests/css3/filters/backdrop/backdropfilter-property-parsing-expected.txt: Updated.
* LayoutTests/css3/filters/backdrop/backdropfilter-property-parsing.html: Removed
unnecessary test of getPropertyCSSValue.

* LayoutTests/css3/filters/unprefixed-expected.txt: Updated.
* LayoutTests/css3/filters/unprefixed.html: Use getPropertyValue, shouldBeEqualToString,
and js-test.js.

* LayoutTests/css3/font-variant-parsing-expected.txt: Updated.
* LayoutTests/css3/font-variant-parsing.html: Rewrote test to make it clear what each
line is testing. Use getPropertyValue and js-test.js.

Canonical link: <a href="https://commits.webkit.org/295418@main">https://commits.webkit.org/295418@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/93b787e157c39f38fccb0748dbb8981c949d885e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105011 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24725 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15147 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110228 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55691 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25135 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33269 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79742 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108017 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19563 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94778 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60049 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19312 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12856 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55068 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89030 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12901 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112730 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32176 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23680 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88824 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32541 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91003 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88453 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22554 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33344 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11130 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/27534 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32100 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31893 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35234 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33452 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/93b787e157c39f38fccb0748dbb8981c949d885e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45904

annevk reviewed (approved):
Nice!

darinadler committed:
Eliminate use of getPropertyCSSValue from tests, part 1
https://bugs.webkit.org/show_bug.cgi?id=293570
rdar://152021244

Reviewed by Anne van Kesteren.

* LayoutTests/animations/fill-mode-reverse.html: Use parseFloat and getPropertyValue.

* LayoutTests/animations/spring-computed-style-expected.txt: Updated.
* LayoutTests/animations/spring-computed-style.html: Use getPropertyValue and js-test.js.

* LayoutTests/animations/spring-parsing-expected.txt: Updated.
* LayoutTests/animations/spring-parsing.html: Removed unnecessary test of getPropertyCSSValue.
Use shouldBeEqualToString and js-test.js.

* LayoutTests/animations/unprefixed-properties-expected.txt: Updated.
* LayoutTests/animations/unprefixed-properties.html: Removed unnecessary test of
getPropertyCSSValue. Use shouldBeEqualToString and js-test.js.

* LayoutTests/animations/unprefixed-shorthand-expected.txt: Updated.
* LayoutTests/animations/unprefixed-shorthand.html: Use getPropertyValue,
shouldBeEqualToString, and js-test.js.

* LayoutTests/css3/calc/transitions-dependent.html: Renamed function in test to
propertyValue instead of getPropertyCSSValue: it has no relationshiop with the
getPropertyCSSValue we are trying to get rid of.

* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled.html: Removed unnecessary
test of getPropertyCSSValue.

* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled.html: Removed unnecessary
test of getPropertyCSSValue.

* LayoutTests/css3/filters/backdrop/backdropfilter-property-computed-style-expected.txt
Updated.
* LayoutTests/css3/filters/backdrop/backdropfilter-property-computed-style.html:
Use getPropertyValue, shouldBeEqualToString, and js-test.js.

* LayoutTests/css3/filters/backdrop/backdropfilter-property-parsing-expected.txt: Updated.
* LayoutTests/css3/filters/backdrop/backdropfilter-property-parsing.html: Removed
unnecessary test of getPropertyCSSValue.

* LayoutTests/css3/filters/unprefixed-expected.txt: Updated.
* LayoutTests/css3/filters/unprefixed.html: Use getPropertyValue, shouldBeEqualToString,
and js-test.js.

* LayoutTests/css3/font-variant-parsing-expected.txt: Updated.
* LayoutTests/css3/font-variant-parsing.html: Rewrote test to make it clear what each
line is testing. Use getPropertyValue and js-test.js.

Canonical link: https://commits.webkit.org/295418@main

darinadler commented:
I realize now after landing this that this does remove some test coverage that could be useful to keep `getPropertyCSSValue` tested as long as it’s web-exposed. I wonder what the best way to handle that is.

Some tests are arbitrarily using `getPropertyCSSValue` and would be unambiguously better if they didn’t use it. Even those are accidentally testing some `CSSValue` code, so in some cases changing them to not use `getPropertyCSSValue` has a small cost in test coverage.

Other tests are explicitly testing `getPropertyCSSValue` alongside other things, but often in not a particularly useful or thorough way. I’ve been removing those tests and now I am not sure I should be!

Still other tests are explicitly about `CSSValue` objects, including tests like the wrapper tests. Those clearly shouldn’t be removed until we aren’t getting any useful test coverage out of them.

One possibility is to keep testing `CSSValue`, but put all the tests in one directory inside LayoutTests rather than having them spread around. Another option would be to have `CSSValue` tests use functions in internals, so the tests would keep working even when we change `getPropertyCSSValue` to not be web-exposed.

darinadler commented:
Also, some of the tests duplicate tests in WPT, in fact some of those WPT tests might have been derived from the WebKit tests at some point. It’s possible that the `CSSValue` parts of these tests are the only usefully different parts, and so in those cases I should probably be removing the tests entirely rather than removing the only thing that makes them different.
